core/common/src/main/java/alluxio/security/authentication/AuthenticatedUserInjector.java

```diff
@@ -102,13 +104,31 @@ public void onHalfClose() {
       } catch (UnauthenticatedException e) {
         String message = String.format("Channel: %s is not authenticated for call: %s",
             channelId.toString(), call.getMethodDescriptor().getFullMethodName());
-        call.close(Status.UNAUTHENTICATED.withDescription(message), headers);
+        safeClose(call, Status.UNAUTHENTICATED.withDescription(message), headers);
       }
     } else {
       String message = String.format("Channel Id is missing for call: %s.",
           call.getMethodDescriptor().getFullMethodName());
-      call.close(Status.UNAUTHENTICATED.withDescription(message), headers);
+      safeClose(call, Status.UNAUTHENTICATED.withDescription(message), headers);
     }
     return callAuthenticated;
   }
+
+  /**
+   * Closes the call while blanketing runtime exceptions.
+   * This is mostly to avoid dumping "already closed" exceptions to logs.
+   *
+   * @param call call to close
+   * @param status status to close the call with
+   * @param headers headers to close the call with
+   */
+  private <ReqT, RespT> void safeClose(ServerCall<ReqT, RespT> call, Status status,
+      Metadata headers) {
+    try {
+      call.close(status, headers);
```

Comment:
Add a debug logging message for the authentication error on the server side?

Reply:
Could you clarify your ask? (This is server side)

Reply:
If that's what you means; "Status: {}" in the below debug print will contain with what the call was being closed for.

Reply:
Right now the authentication error message is only reported to the client side through the Status object. We should consider adding debug logging on the server side so even if we don't have the client side message we still have a way to know what went wrong with authentication. This should be logged regardless whether the close of the call result in an exception or not.

Reply:
Got it. Added trace before closing the call.